Lint Clean Up

MMurrayAtBR opened the pull request:
Updating of files to pass lint checker and removal of unnecessary Suppress annotations

Commit:
Updating of files for lint check warnings

jaredstockton commented on app/src/main/res/drawable/ic_home.xml:
```diff
@@ -1,14 +1,18 @@
-        android:fillColor="#E2231A"
-        android:fillType="nonZero"/>
-  </group>
+    <!-- "clip-path" attribute not supported API < 24; Check that removing
+    this path data does not affect the look of the graphic -->
```
@MMurrayAtBR This comment can be removed now that the minSdkVersion is set to 24

jaredstockton commented on buildSrc/src/main/kotlin/Dependencies.kt:
```diff
@@ -89,7 +89,7 @@ object Config {
         // https://developer.android.com/studio/releases/build-tools
         const val BUILD_TOOLS = "31.0.0"
-        const val MIN_SDK = 23 // TODO: TEMPLATE - Replace with appropriate project minSdkVersion
+        const val MIN_SDK = 24 // TODO: TEMPLATE - Replace with appropriate project minSdkVersion
```
@MMurrayAtBR While running `lintInternalDebug` locally, it was mentioned that there was a `drawable-v24` resource . That resource should actually be deleted as it uses an incorrect app icon for v24+ devices. The existing resource in `drawable` is the preferred drawable.

- [x] Delete `ic_launcher_foreground.xml` from `drawable-v24`

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt:
```diff
@@ -63,6 +63,8 @@ fun AuthCodePreview() {
     }
 }
 
+// FIX ME: javaScriptEnabled = true can result in removal from Play Store
```
@MMurrayAtBR nit: Change `FIX ME` `FIXME` to follow the naming used elsewhere in the project. Additionally, using `FIXME` allows project-wide searches to show all instances as well as cause all instances to show in the TODO view of Android Studio.

```suggestion
// FIXME: javaScriptEnabled = true can result in removal from Play Store
```

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt:
```diff
@@ -14,12 +13,12 @@ object Routes {
     const val Profile = "profile"
 
-    fun RepositoryBrowser(data: RepositoryBrowserData) =
+    fun repositoryBrowser(data: RepositoryBrowserData) =
```
@BottleRocket-Colin I understand why you used UpperCamelCase for these function names (mimicking the approach use for Composable functions). I don't really have a strong opinion for or against them so would appreciate to hear your feedback. Have you seen examples of Google / other sample repositories using UpperCamelCase style for routes?

cc: @MMurrayAtBR

BottleRocket-Colin replied:
Good Catch!.  @MMurrayAtBR Let's change these back to UpperCamelCase and add a suppress annotation.  That way they will be consistent with the routes above.

jaredstockton commented on app/src/main/res/values/strings.xml:
```diff
@@ -19,4 +19,18 @@
     <!-- Misc -->
     <string name="dev_options_button">Dev Options</string>
 
+    <string name="today">Today</string>
```
@MMurrayAtBR Why are these necessary in the app module?

MMurrayAtBR replied:
These string resources are being used in the app module "testResources" & "testContext" mocks, but were not in the app module string resources. 
I chose to add the missing strings to the app module res file & resolve the Unresolved References, instead of making changes to the mocks

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawer.kt:
```diff
@@ -124,24 +124,28 @@ fun ProfileScreenPreview() {
-    )
-)
+fun generateItemListPreview(): List<NavItemState> {
+    val itemList by remember {
```
@MMurrayAtBR This approach seems fine to me.

@BottleRocket-Colin Thoughts?

BottleRocket-Colin replied:
Works for me.

jaredstockton reviewed (commented):
@MMurrayAtBR Thanks for the PR! I have left a few comments to address prior to approval :)

Commit:
Unit Tests [updates]

- Fixed the "FIXME" comment on line 66
- Removed now unnecessary comment about "clip-path" in ic_home
- Deleted drawable-v24 ic_launcher_foreground.xml
- Capitalized functions "RepositoryBrowser()" and "RepositoryFile()"

jaredstockton reviewed (approved):
@MMurrayAtBR Thanks for making the changes! Looks good now :)
